app/jobs/generate_uniprot_mapping_tsv.rb

```diff
@@ -0,0 +1,54 @@
+class GenerateUniprotMappingTsv < GenerateTsvs
+    def perform
+      ensure_downloads_directory_exists
+      tsvs_to_generate.each do |e|
+        begin
+          tmp_file = tmp_file(e.file_name)
+          tmp_file.puts(e.headers.join("\t"))
+  
+          e.objects.find_each do |object|
+            
+            row = e.row_from_object(object)
+            if row[1].is_a?(Array)
```

Comment:
I might change this to be `e.rows_from_object()` and assume you always get back an Array of rows and push the logic of handling multiple (or no) uniprot ids down a level. (See other comment)